src/ShopingCenter.java

```diff
@@ -0,0 +1,150 @@
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ShopingCenter {
+
+// *********** Static fields **********
+
+    private static int nextId = 0;
+
+// *********** End of Static fields ***********
+
+// *********** Fields ************
+
+    private String name;
+    private Address address;
+    private List<Shop> shopList;
+    private OpeningHours timeOpen;
+    private int id;
+
+    public ShopingCenter(String name, String address, int openTime, int closeTime) {
+        this.name = name;
+        this.address = new Address(address);
+        this.shopList = new LinkedList<Shop>();
+        this.timeOpen = new OpeningHours(openTime, closeTime) ;
+        this.id = ShopingCenter.nextId++;
+    }
+
+// ********** End of fields ************
+
+// ********** Constructors for ShoppingCenter ***********
+
+    public ShopingCenter(String name, Address address, OpeningHours timeOpen) {
+        this.name = name;
+        this.address = address;
+        this.shopList = new LinkedList<Shop>();
+        this.timeOpen = timeOpen;
+        this.id = ShopingCenter.nextId;
```

Comment:
compare line 28 and 40, similar, but not identical

Reply:
Again ++